packages/workspace/src/affine/sync.ts

```diff
@@ -0,0 +1,87 @@
+import { CRUD } from '@affine/workspace/affine/crud';
+import { createCloudBlobStorage } from '@affine/workspace/blob/cloud-blob-storage';
+import { downloadBinaryFromCloud } from '@affine/workspace/providers/cloud';
+import { assertExists } from '@blocksuite/global/utils';
+import { createIndexeddbStorage } from '@blocksuite/store';
+import type { Doc } from 'yjs';
+import { applyUpdate } from 'yjs';
+
+let abortController: AbortController | undefined;
+
+const downloadRecursive = async (
+  rootGuid: string,
+  doc: Doc,
+  signal: AbortSignal
+): Promise<void> => {
+  if (signal.aborted) {
+    return;
+  }
+  const binary = await downloadBinaryFromCloud(rootGuid, doc.guid);
+  if (typeof binary !== 'boolean') {
+    applyUpdate(doc, new Uint8Array(binary), 'affine-cloud-service');
```

Comment:
I am a little worried that applying update here will cause performance issues for the foreground tasks. is it possible to move to worker instead?

Reply:
Fixed